fapi-client/src/main/scala/com/gu/facia/api/FAPI.scala

```diff
@@ -189,4 +190,87 @@ object FAPI {
+    case supportingCuratedContent: SupportingCuratedContent => supportingCuratedContent.copy(properties = properties)
+    case snap: LatestSnap => snap.copy(properties = properties)
+    case snap: LinkSnap => snap.copy(properties = properties)
+  }
+
+  private def updateGroup(content: FaciaContent, group: String): FaciaContent = content match {
+    case curatedContent: CuratedContent => curatedContent.copy(group = group)
+    case supportingCuratedContent: SupportingCuratedContent => supportingCuratedContent.copy(group = group)
+    case snap: LatestSnap => snap.copy(group = group)
+    case snap: LinkSnap => snap.copy(group = group)
+  }
+
+  private def applyDefaultBoost[T](
+    content: T,
+    default: BoostLevel,
+    allowedBoosts: List[BoostLevel],
+    getBoostLevel: T => BoostLevel,
+    setBoostLevel: (T, BoostLevel) => T
+  ): T =
+    if (allowedBoosts.contains(getBoostLevel(content)))
+      content
+    else
+      setBoostLevel(content, default)
+
+  case class GroupBoostConfig(default: BoostLevel, allowedBoosts: List[BoostLevel], maxItems: Int, group: String)
+
+  private val flexibleGeneralBoosts = List(
+    // Splash
+    GroupBoostConfig(BoostLevel.Default, List(BoostLevel.Default, BoostLevel.Boost, BoostLevel.MegaBoost, BoostLevel.GigaBoost), maxItems = 1, group = "3"),
+    // Very large
+    GroupBoostConfig(BoostLevel.MegaBoost, List(BoostLevel.MegaBoost), maxItems = 0, group = "2"),
+    // Large
+    GroupBoostConfig(BoostLevel.Boost, List(BoostLevel.Boost, BoostLevel.MegaBoost), maxItems = 0, group = "1"),
+    // Standard
+    GroupBoostConfig(BoostLevel.Default, List(BoostLevel.Default, BoostLevel.Boost, BoostLevel.MegaBoost), maxItems = 8, group = "0")
+  )
+
+  private def boostsConfigFor(collectionType: String, groupsConfig: List[GroupConfig]): Option[List[GroupBoostConfig]] = {
+    val groupsConfigBiggestFirst = groupsConfig.reverse
+    condOpt(collectionType) {
+      case "flexible/general" => flexibleGeneralBoosts.zip(groupsConfigBiggestFirst).map { case (boosts, groupConfig) =>
+        boosts.copy(maxItems = groupConfig.maxItems.getOrElse(boosts.maxItems))
+      }
+    }
+  }
+
+  def applyDefaultBoostLevelsAndGroups(collectionConfig: CollectionConfig, contents: List[FaciaContent]): List[FaciaContent] = {
+    applyDefaultBoostLevelsAndGroups[FaciaContent](
+      groupsConfig = collectionConfig.groupsConfig,
+      collectionType = collectionConfig.collectionType,
+      contents = contents,
+      getBoostLevel = _.properties.boostLevel,
+      setBoostLevel = (content, default) => updateContentProperties(content, content.properties.copy(boostLevel = default)),
+      setGroup = updateGroup
+    )
+  }
+
+  // NB - this will not behave as expected if there are "gaps" in the curated content (ie there are no curated cards in the
+  // second group but there are curated cards in the third group). But the fronts tool is supposed to prevent gaps.
```

Comment:
I'm not sure we've got this implemented at the moment so maybe something we need to add there? As in it's possible to have a flexible general with nothing in a group - is this the kind of situation that would lead to unexpected behaviour? 
![image](https://github.com/user-attachments/assets/45395aad-bc32-4fc7-8939-c9fdfcb5f67a)

Reply:
I've discussed with Jonathan, and while it isn't ideal that the fronts tool doesn't bump things up, it is preferable that we get this change out in the platforms to support backfill